api/MediaMetadata.json

```diff
@@ -13,18 +13,24 @@
           "edge": {
             "version_added": false
           },
-          "firefox": {
-            "version_added": "71",
-            "flags": [
-              {
-                "type": "preference",
-                "name": "dom.media.mediasession.enabled",
-                "value_to_set": "true"
-              }
-            ]
-          },
+          "firefox": [
+            {
+              "version_added": "82"
+            },
+            {
+              "version_added": "71",
+              "flags": [
+                {
+                  "type": "preference",
+                  "name": "dom.media.mediasession.enabled",
+                  "value_to_set": "true"
+                }
+              ]
+            }
+          ],
           "firefox_android": {
-            "version_added": null
+            "version_added": "82",
+            "notes": "The particular Media Session used for Android media control depends on the user agent. The Android browser implementation does not allow any session to be selected as active."
```

Comment:
I'm having a hard time understanding this note. I ended up with a bunch of questions.

What does "depends on the user agent" mean here? Isn't Firefox the user agent? What about "does not allow any session to be selected as active"? This feature is the `MediaMetadata` interface itself. How does one use the interface to "select" something?

I need some help here to figure out how to translate this into something that a web developer can act on. `MediaMetadata` apparently has some limitation on Android. As developer, what can I do with it? What can't I do with it? How is it like other implementations? How is it different?

I see now that this note appears in several places. When is it most important to developers? When are they most likely to notice the behavior on Android?

Reply:
@ddbeck What this says is that the API has been implemented but really can't be used. However it code running in a page that uses this API will never know that. Explanation:

The theory is that pages use this API to be able to expose media data and control to a user agent. In theory a user should be able to access an interface to select from pages using the API (and other media sources) and set one as being used by standard mechanisms for media display/control on the device. In other words, if your device supports playing media from the lockscreen (and displaying the current song say) then you should be able to select one of the pages as the source for media.

But what has happened is that the API was implemented, but the components required to make that UI selection to assign a page to the lockscreen (say) do not work/exist. So the API "works" - you can write a page that uses it. But then you still can't select the page.

An subtle point here though is that many pages might use the API, and most of them will not be selected by the user agent. No page actually knows if it is being used.  There is nothing a developer can do about this. 

The text is what the developer thought made sense to cover this. It is strictly accurate. But perhaps we should be blunt and say - there is no point using this?

Maybe something more like (?):

> At time of writing the Android browser implementation does not allow web pages to be set as a media source. This API can therefore be used, but not tested.

Reply:
Ah, OK. Thank you, @hamishwillee! That's really helpful. I think this helps me understand what's going on here a bit better. I think we need to revise the note and tweak the data a bit.

I think your draft note is pretty good start. To help this note standalone better in multiple contexts (MDN's not the only BCD consumer), does something like this revision work?

> Firefox does not implement a user interface for choosing a web page as a media source. This API is exposed, but has no corresponding user-facing behavior.

That seems like a big enough caveat that we'd also want to set `"partial_implementation": true`.

Reply:
> Firefox does not implement a user interface for choosing a web page as a media source. This API is exposed, but has no corresponding user-facing behavior.

I will go back and check with engineering because it may be that the missing part of the picture is outside of the browser/in the OS; in which case this feels wrong. That then would also make `"partial_implementation": true` feel wrong.

Reply:
@ddbeck An excellent description of the [integration added here](https://bugzilla.mozilla.org/show_bug.cgi?id=1665496#c13). In summary, it is as we thought - this API is integrated into GeckoView, but anything that uses GeckoView must further integrate with the underlying platform via the embedding application - in this case Firefox Android.

I have therefore taken your advice and changed to partial implementation and added note as below in all locations.

> Firefox does not yet integrate this API into Android's media control interfaces. Hence the API is exposed, but the corresponding user-facing behavior is not available.

Should be good to review again.